local-volume/provisioner/cmd/main.go

```diff
@@ -18,18 +18,32 @@ package main
 
 import (
 	"flag"
+	"log"
+	"net/http"
 	"os"
 
 	"github.com/golang/glog"
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/common"
 	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/controller"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/deleter"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/metrics"
+	"github.com/kubernetes-incubator/external-storage/local-volume/provisioner/pkg/metrics/collectors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
 
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 )
 
+var (
+	optListenAddress string
+	optMetricsPath   string
+)
+
 func main() {
+	flag.StringVar(&optListenAddress, "listen-address", ":8080", "address on which to expose metrics")
```

Comment:
We've been putting all our configuration parameters into a configmap, instead of command line arguments. @dhirajh do you have any preference?

Reply:
@msau42 Looks like I missed this comment, sorry about the delayed response. ConfigMap would have been ideal as one does not need to update docker image or mess around with entry point overrides in the manifest to change the CLI option. cc @cofyc

Reply:
I prefer CLI option. If it's configured via ConfigMap, a few issues we have to consider:

- on port change, should we restart listen goroutine to listen on new port?
- if we should re-listen on new port, should it fail if there is port conflict?

IMO, if we need to configure listen address (in rare cases i think), it's simpler to update deployment or damonset (and delete pods) to restart pods again. If there is port conflict, pods will fail and we will know it. This is normal operation in kubernetes.